src/main/java/alcazar/Deadline.java

```diff
@@ -0,0 +1,43 @@
+package alcazar;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+public class Deadline extends Task {
+
+    private String by;
+    private String d1;
+
+    int i = 0;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by.trim();
+        String date = "";
+        for (i = 0; i < by.length(); i++) {
+            if (by.charAt(i) == ' ') {
+                break;
+            }
+            date += by.charAt(i);
+        }
```

Comment:
```suggestion
    public Deadline(String description, String by) {
        super(description);
        this.by = by.trim();
        String date = "";
        for (int i = 0; i < by.length(); i++) {
            if (by.charAt(i) == ' ') {
                break;
            }
            date += by.charAt(i);
        }
```

Should this variable be declared within the constructor instead?